docs/formatter.md

```diff
@@ -0,0 +1,251 @@
+indent-style = "tab"
+```
+
+For more on configuring Ruff via `pyproject.toml`, see [_Configuring Ruff_](configuration.md).
+
+Given the focus on Black compatibility (and unlike formatters like [YAPF](https://github.com/google/yapf)),
+Ruff does not currently expose any configuration options to modify core formatting behavior outside
+of these trivia-related settings.
+
+## Format suppression
+
+Like Black, Ruff supports `# fmt: on`, `# fmt: off`, and `# fmt: skip` pragma comments, which can
+be used to temporarily disable formatting for a given code block.
+
+`# fmt: on` and `# fmt: off` comments are enforced at the statement level:
+
+```python
+# fmt: off
+not_formatted=3
+also_not_formatted=4
+# fmt: on
+```
+
+As such, adding `# fmt: on` and `# fmt: off` comments within expressions will have no effect. In
+the following example, both list entries will be formatted, despite the `# fmt: off`:
+
+```python
+[
+    # fmt: off
+    '1',
+    # fmt: on
+    '2',
+]
+```
+
+Instead, apply the `# fmt: off` comment to the entire statement:
+
+```python
+# fmt: off
+[
+    '1',
+    '2',
+]
+# fmt: on
+```
+
+`# fmt: skip` comments suppress formatting for a preceding statement, case header, decorator,
+function definition, or class definition:
+
+```python
+if True:
+    pass
+elif False: # fmt: skip
+    pass
+
+@Test
+@Test2 # fmt: off
+def test(): ...
+
+a = [1, 2, 3, 4, 5] # fmt: off
+
+def test(a, b, c, d, e, f) -> int: # fmt: skip
+    pass
+```
+
+Like Black, Ruff will _also_ recognize [YAPF](https://github.com/google/yapf)'s `# yapf: disable` and `# yapf: enable` pragma
+comments, which are treated equivalently to `# fmt: off` and `# fmt: on`, respectively.
+
+## Conflicting lint rules
+
+Ruff's formatter is designed to be used alongside the linter. However, the linter includes
+some rules that, when enabled, can cause conflicts with the formatter, leading to unexpected
+behavior.
+
+When using Ruff as a formatter, we recommend disabling the following rules:
+
+- [`line-too-long`](rules/line-too-long.md) (`E501`)
+- [`single-line-implicit-string-concatenation`](rules/single-line-implicit-string-concatenation.md) (`ISC001`)
+- [`missing-trailing-comma`](rules/missing-trailing-comma.md) (`COM812`)
+- [`prohibited-trailing-comma`](rules/prohibited-trailing-comma.md) (`COM819`)
+- [`bad-quotes-inline-string`](rules/bad-quotes-inline-string.md) (`Q000`)
+- [`bad-quotes-multiline-string`](rules/bad-quotes-multiline-string.md) (`Q001`)
+- [`bad-quotes-docstring`](rules/bad-quotes-docstring.md) (`Q002`)
+- [`avoidable-escaped-quote`](rules/avoidable-escaped-quote.md) (`Q003`)
+
+Similarly, we recommend disabling the following isort settings, which are incompatible with the
+formatter's treatment of import statements when set to non-default values:
+
+- [`lines-after-imports`](https://docs.astral.sh/ruff/settings/#isort-lines-after-imports)
+- [`lines-between-types`](https://docs.astral.sh/ruff/settings/#isort-lines-between-types)
```

Comment:
Our [internal configuration document](https://www.notion.so/astral-sh/Configuration-c291a149e50f438083221a65468a736f?pvs=4#972f1a2e4ce14549b899adce6563b931) mentions more conflicting rules and isort options.

Reply:
I'll add them, but it's the same confusion as in your PR that enables the warnings -- what constitutes a "conflict"?

Reply:
I also added `ISC001` and `Q003`, were those intentionally omitted?